### DISCUSSION TITLE
Overriding included tasks

### DISCUSSION THREAD

**Discussion by samegutt** [psake/psake#380] [Q&A]
Given the two scripts `CommonTasks.ps1`

``` powershell
Task Build -depends Clean, Compile

Task Clean{ }

Task Compile { }
```

and `MyTasks.ps1`

``` powershell
Include ".\CommonTasks.ps1"

task default -depends Build

task MyClean -alias "Clean" { }
```

The task `MyClean` effectively overrides the task `Clean` with its alias, giving the following output:

``` powershell
C:\> Invoke-psake .\MyTasks.ps1
Executing MyClean
Executing Compile

Build Succeeded!
```

Q: It this by design? Or is this unintentional?  I am creating some common build steps for the project I work for, and some of the modules needs some customisation to the build flow. Using the `-alias` parameter I get the functionality I need, but the solution has some apparent smells.

Suggestion: An `-override` parameter would make the purpose of `MyClean`very clear.

**Comment by UberDoodles** [comment]
@samegutt 

I just posted a reply to #298, and I realised my reply might also apply here. Interestingly, that issue suggested the ability to merge tasks, rather than override them.

Generally, I don't define dependencies for any tasks which have an action. When I want to chain tasks together; I create a separate task for that purpose. It gives you more flexibility to run exactly what you want to. 

Would that help you achieve what you are trying to do?

There might be other use cases where being able to override (or merge) tasks is useful though, and I like your suggestion about adding an -override parameter to make it clear what is going to happen to the two task definitions.

**Comment by HeyItsGilbert** [comment]
The opening question here is "is this by design?", which makes it a better fit for Discussions than the issue tracker, so I'm moving it there. The short answer: aliasing-overrides-the-original isn't a documented or supported pattern, so I wouldn't lean on it. #298 raised the inverse idea - merging tasks rather than overriding them - which is worth folding into the same discussion if you want to push the override use case forward.
